kedro/framework/cli/utils.py

```diff
@@ -401,68 +402,33 @@ def _reformat_load_versions(  # pylint: disable=unused-argument
     return load_versions_dict
 
 
-def _try_convert_to_numeric(value):
-    try:
-        value = float(value)
-    except ValueError:
-        return value
-    return int(value) if value.is_integer() else value
-
-
 def _split_params(ctx, param, value):
     if isinstance(value, dict):
         return value
-    result = {}
+    dot_list = []
     for item in split_string(ctx, param, value):
-        item = item.split(":", 1)
-        if len(item) != 2:
+        equals_idx = item.find("=")
+        colon_idx = item.find(":")
+        if equals_idx != -1 and colon_idx != -1 and equals_idx < colon_idx:
```

Comment:
While updating the docs I realised that the original implementation would not work for cases where both colons and equal signs are used. For example : `--params url=http://example.com`. This if statement checks for this specific case. I've also added these examples to the tests.

Reply:
It might be useful to add a comment in the code as well to explain what this check is for.